Mathlib/NumberTheory/ModularForms/EisensteinSeries/QExpansion.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2025 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.Analysis.Complex.SummableUniformlyOn
+import Mathlib.Analysis.SpecialFunctions.Trigonometric.Cotangent
+
+/-!
+# Einstein series q-expansions
+
+We give some identities for q-expansions of Eisenstein series that will be used in describing their
+q-expansions.
+
+-/
+
+open Set Metric TopologicalSpace Function Filter Complex EisensteinSeries
+
+open _root_.UpperHalfPlane hiding I
+
+open scoped Topology Real Nat Complex Pointwise
+
+local notation "ℍₒ" => upperHalfPlaneSet
+
+lemma iteratedDerivWithin_cexp_mul_const (k m : ℕ) (p : ℝ) {S : Set ℂ} (hs : IsOpen S) :
```

Comment:
The name is not really conforming. In any case, I think this is rather specific for the intended application, and so is perhaps better as a `private` lemma (it is a minor variation on `iteratedDeriv_cexp_const_mul` and does not look like it might be useful outside the context of this file).